### PR TITLE
Updated arm cortex status

### PR DESCRIPTION
**Pull request by rrobgill**
[Arm Developer](https://developer.arm.com/support/arm-security-updates/speculative-processor-vulnerability)
A8 & A12 are vulnerable to Variants 1&2.
All arch 8 cortexes A57-A76 are vulnerable to variant 4.

Applies all information about vulnerabilities of ARM Cortex processors (from https://developer.arm.com/support/arm-security-updates/speculative-processor-vulnerability).

Whitelist & blacklist approach, using both vulnerable and non vulnerable status for each identified CPU, with vulnerabilities tracked cumulatively for multi CPU systems.

(Fixes false non-vulnerable status for A8 reported by @V10lator in #206  )

### CODE REVIEW

File: spectre-meltdown-checker.sh [hunk 1]
```diff
@@ -345,63 +345,58 @@ is_cpu_vulnerable()
...
 	fi
-	# from the information we have for now, it seems that CPUs that are vulnerable to variant1 are also vulnerable to variant4
-	[ -z "$variant4" ] && variant4=$variant1
-	_debug "is_cpu_vulnerable: temp results are <$variant1> <$variant2> <$variant3> <$variant3a> <$variant4>"
-	# if at least one of the cpu is vulnerable, then the system is vulnerable
-	[ "$variant1"  = "immune" ] && variant1=1  || variant1=0
-	[ "$variant2"  = "immune" ] && variant2=1  || variant2=0
-	[ "$variant3"  = "immune" ] && variant3=1  || variant3=0
-	[ "$variant3a" = "immune" ] && variant3a=1 || variant3a=0
-	[ "$variant4"  = "immune" ] && variant4=1  || variant4=0
```
Comment (speed47) [inline spectre-meltdown-checker.sh]: Thanks for the updates regarding the ARM models!

However lines 398-404: why did you delete them? Their job is to convert the temporary values "immune" or "vuln" or "" into 0 or 1 that is used and expected in the remaining of the script (those lines are executed for all archs, not only ARM).

Comment (rrobgill) [reply spectre-meltdown-checker.sh]: I've just unbroken that bit.

**Comment by SvOlli** [comment]
The commit 181978d7c377bfdbd5eea6bc19dbd98450218af broke something. Now all ARM CPUs immune to spectre/meltdown like A7 (CPU part: 0xc07, A53 (CPU part: 0d03) are claimed to be vulnerable.
